pkg/containertools/runner.go

```diff
@@ -121,34 +125,73 @@ func (r *ContainerCommandRunner) Build(dockerfile, tag string) error {
 	return nil
 }
 
-// Unpack copies a directory from a local container image to a directory in the local filesystem.
+// Unpack copies a file or directory from a local container image to a directory on the local filesystem.
 func (r *ContainerCommandRunner) Unpack(image, src, dst string) error {
-	args := r.argsForCmd("create", image, "")
-
-	command := exec.Command(r.containerTool.String(), args...)
+	ctx, cancel := context.WithCancel(context.TODO())
+	defer cancel()
 
+	args := r.argsForCmd("create", image, "")
+	command := exec.CommandContext(ctx, r.containerTool.String(), args...)
 	r.logger.Infof("running %s create", r.containerTool)
 	r.logger.Debugf("%s", command.Args)
 
 	out, err := command.CombinedOutput()
 	if err != nil {
 		r.logger.Errorf(string(out))
-		return fmt.Errorf("error creating container %s: %v", string(out), err)
+		return fmt.Errorf("error creating container %s: %w", string(out), err)
+	}
+
+	// Create a new untarer to copy and read output from stdout to a tar reader.
+	// This is done to rewrite file permissions on the fly to avoid permissions-related errors in standard docker cp.
+	piper, pipew := io.Pipe()
+	defer func() {
+		if err := piper.Close(); err != nil {
+			r.logger.Warnf("error closing command read pipe: %s", err)
+		}
+	}()
+	defer func() {
+		if err := pipew.Close(); err != nil {
+			r.logger.Warnf("error closing command write pipe: %s", err)
+		}
+	}()
+
+	if err != nil {
+		return fmt.Errorf("error setting unpacking destination: %w", err)
```

Comment:
What error is being caught here? The error captured in the defer statements?

Reply:
ah, this is a leftover from refactoring. nice catch!

Reply:
fixed